server/server/src/sync/remote_data_synchroniser.rs

```diff
@@ -309,6 +306,13 @@ impl<'a> RemoteSyncState<'a> {
     }
 
     pub fn set_initial_remote_data_synced(&self) -> Result<(), RepositoryError> {
```

Comment:
btw. I had a chat with Roxy this morning and we thought its best to also set the site_id in the method, i.e. introduce a new key value store entry for the used site_id so that we know which site id has been used for the sync. This could be used to verify that a config doesn't use a different site_id...

(problem was that SettingsSyncSideId might not always be set, at least in our current setup)

Reply:
> This could be used to verify that a config doesn't use a different site_id

It's very possible for the same site id to be use for different sites but with different central data files. 

I think site_id should be set prior to this method running, as per previous comment, on startup (if server is started with sync settings in env vars or config fie), the reason for this: Some sites will take awhile to initialise, if there is connection error during initialisation, it would be good if it can be resumed